### PR TITLE
Ft account activation #187419176

### PR DESCRIPTION
**Pull request by MugemaneBertin2001**
# what does this PR do?
- Enables an admin to activate and deactivate user account 
# description of task to be complete
- Due to varying reasons e.g fraud, abuse, malicious use or violation of terms of service, admin should be able to diable any user account.

- An admin should be able to activate / reactive an account.
for deactivation, send an email to the user with reasons for deactivation

## How should this be manually tested?
1. checkout this branch and run app
2. create a user
3.  patch /api/v1/users/{userId}/status

Screenshot 2024-05-06 180118
Screenshot 2024-05-06 180218

# pivotal tracker id ---->#187419176

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+            "server": "localhost",
+            "port": 5432,
+            "driver": "PostgreSQL",
+            "name": "now",
+            "group": "now",
+            "database": "postgres",
+            "username": "postgres"
+        }
+    ]
+}
```
Comment (teerenzo) [inline .vscode/settings.json]: remove this

File: src/services/user.service.ts [hunk 2]
```diff
@@ -158,3 +164,39 @@ export const updateUserRoleService = async (userId: number, newRoleId: number):
...
+    await sendEmailService(updatedUser[0], subject, template);
+    return {
+      status: 200,
+      data: {
+        user: updatedUser,
+        message: 'User account status updated successfully',
+      },
+    };
+  } catch (error:any) {
+    console.error('Error updating user account status:', error.message);
```
Comment (teerenzo) [inline src/services/user.service.ts]: remove this console

Comment (teerenzo) [inline src/services/user.service.ts]: remove this console

**Review by teerenzo — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+            "server": "localhost",
+            "port": 5432,
+            "driver": "PostgreSQL",
+            "name": "now",
+            "group": "now",
+            "database": "ecommerce_db",
+            "username": "postgres"
+        }
+    ]
+}
```
Comment (teerenzo) [inline .vscode/settings.json]: remove it

**Review by Heisjabo — APPROVED** [review]

**Review by soleil00 — APPROVED** [review]

**Review by yvanddniyo — APPROVED** [review]
look good but Make sure you squash your commit.

File: src/services/user.service.ts [hunk 4]
```diff
@@ -158,3 +165,39 @@ export const updateUserRoleService = async (userId: number, newRoleId: number):
...
+    },
+    {
+      where:{
+      id:userId
+    }})
+    if(!updatedUser){
+      return {
+        status: 500,
+        message:"Internal server error"
+      }
```
Comment (teerenzo) [inline src/services/user.service.ts:184]: this is not server , return other message with different status code

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Review by teerenzo — APPROVED** [review]

**Pull request merged**